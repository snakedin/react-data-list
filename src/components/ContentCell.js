import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class ContentCell extends Component {

    static propTypes = {
        id: PropTypes.string,
        value: PropTypes.func,
        attributes: PropTypes.object
    };

    renderCell() {
        const { id, value, item } = this.props;

        if (value !== 'undefined' && typeof value === 'function') {
            return value(item);
        } else {
            return item[id] !== undefined ? item[id] : null;
        }
    }

    render() {

        const { attributes } = this.props;

        return (
            <td {...attributes}>
                { this.renderCell() }
            </td>
        );
    }
}