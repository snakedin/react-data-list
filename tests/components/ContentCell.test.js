import React from "react";
import { render } from "@testing-library/react";
import ContentCell from "../../src/components/ContentCell";

it("ContentCell render", () => {

  function tdRender(component) {
    const tableRow = document.createElement('tr');

    const { container } = render(component, {
      container: document.body.appendChild(tableRow)
    });

    return container;
  }

  const item = { id: 11, name: "Test Value" };

  const baseRender = tdRender(<ContentCell item={item} id={'name'}/>);
  expect(baseRender).toMatchSnapshot();

  expect(baseRender).toMatchDiffSnapshot(
      tdRender(<ContentCell item={item} id={'none'}/>)
  );

  expect(baseRender).toMatchDiffSnapshot(
      tdRender(<ContentCell item={item} id={'name'} attributes={{'className': 'content-cell'}}/>)
  );

  expect(baseRender).toMatchDiffSnapshot(
      tdRender(<ContentCell item={item} value={(item) => <span>{item.id}: {item.name}</span>}/>)
  );
});
