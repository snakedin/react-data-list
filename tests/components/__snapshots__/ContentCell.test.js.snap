// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCell render 1`] = `
<tr>
  <td>
    Test Value
  </td>
</tr>
`;

exports[`ContentCell render 2`] = `
"Snapshot Diff:
- First value
+ Second value

  <tr>
-   <td>
-     Test Value
-   </td>
+   <td />
  </tr>"
`;

exports[`ContentCell render 3`] = `
"Snapshot Diff:
- First value
+ Second value

  <tr>
-   <td>
+   <td
+     class=\\"content-cell\\"
+   >
      Test Value
    </td>
  </tr>"
`;

exports[`ContentCell render 4`] = `
"Snapshot Diff:
- First value
+ Second value

  <tr>
    <td>
+     <span>
+       11
+       : 
        Test Value
+     </span>
    </td>
  </tr>"
`;
